### PR TITLE
feature: update prometheus stack

### PR DESCRIPTION
**Pull request by Deepzima**
- Upgraded prometheus operator to v0.62.0
- Upgrade prometheus to v2.41.0
- added release page 2.1.0
- [removed 1.23 support](https://github.com/prometheus-operator/kube-prometheus#compatibility)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,8 @@ Kubernetes Fury Monitoring provides the following packages:
```
Comment (nutellinoit) [inline README.md:46]: Let's not change che main README while we are working on all the packages, we will change this with the last PR

File: docs/releases/v2.1.0.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Monitoring Core Module Release 2.1.0
+
+Welcome to the latest release of the `monitoring` core module of the [`Kubernetes Fury Distribution`](https://github.com/sighupio/fury-distribution), maintained by team SIGHUP.
+
+This is a patch release including the following changes:
+- added `GOMAXPROCS=1` envrionment variable to `node-exporter` to avoid throttling (see [#106](https://github.com/sighupio/fury-kubernetes-monitoring/issues/106)).
+- aligned Service port name for `kube-controller-manager` and `kube-schedule` with the one expected by the ServiceMonitors (see [#109](https://github.com/sighupio/fury-kubernetes-monitoring/issues/109) )
```
Comment (nutellinoit) [inline docs/releases/v2.1.0.md:7]:
```suggestion
TBD
```

File: docs/releases/v2.1.0.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+## Update Guide 🦮
+
+### Process
+
+To upgrade this core module from `v2.0.1` to `v2.1.0`, execute the following:
+
+```shell
+kustomize build vendor/node-exporter | kubectl apply -f --server-side
+kustomize build vendor/kubeadm-sm | kubectl apply -f --server-side
+```
```
Comment (nutellinoit) [inline docs/releases/v2.1.0.md:38]:
```suggestion
TBD
```

File: docs/releases/v2.1.0.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Monitoring Core Module Release 2.1.0
+
+Welcome to the latest release of the `monitoring` core module of the [`Kubernetes Fury Distribution`](https://github.com/sighupio/fury-distribution), maintained by team SIGHUP.
+
+This is a patch release including the following changes:
+- added `GOMAXPROCS=1` envrionment variable to `node-exporter` to avoid throttling (see [#106](https://github.com/sighupio/fury-kubernetes-monitoring/issues/106)).
+- aligned Service port name for `kube-controller-manager` and `kube-schedule` with the one expected by the ServiceMonitors (see [#109](https://github.com/sighupio/fury-kubernetes-monitoring/issues/109) )
+
+> 💡 Please refer to the release notes of the major version [`v2.1.0`](https://github.com/sighupio/fury-kubernetes-monitoring/releases/tag/v2.1.0) if you are upgrading from a version `< v2.0.1`
```
Comment (nutellinoit) [inline docs/releases/v2.1.0.md:9]:
```suggestion
```

File: katalog/prometheus-operated/README.md [hunk 5]
```diff
@@ -15,13 +15,13 @@ Grafana. Grafana integration is provided in Fury monitoring katalog, please see
 
 ## Requirements
 
-- Kubernetes >= `1.21.0`
+- Kubernetes >= `1.23.0`
```
Comment (nutellinoit) [inline katalog/prometheus-operated/README.md:18]:
```suggestion
- Kubernetes >= `1.24.0`
```

Should this be >= 1.24?

File: katalog/prometheus-operator/README.md [hunk 6]
```diff
@@ -39,14 +39,14 @@ illustrated in this image from Prometheus Operator repository:
 
 ## Requirements
 
-- Kubernetes >= `1.21.0`
+- Kubernetes >= `1.23.0`
```
Comment (nutellinoit) [inline katalog/prometheus-operator/README.md:42]:
```suggestion
- Kubernetes >= `1.24.0`
```

Maybe the same applies here

File: katalog/prometheus-operator/kustomization.yaml [hunk 7]
```diff
@@ -3,6 +3,10 @@
 # license that can be found in the LICENSE file.
 
 ---
+# Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
```
Comment (nutellinoit) [inline katalog/prometheus-operator/kustomization.yaml:6]: Is this copyright duplicated?

**Review by nutellinoit — CHANGES_REQUESTED** [review]
Overall LGTM, some changes and questions below 👇

File: katalog/prometheus-operated/prometheusRule.yaml [hunk 8]
```diff
@@ -291,6 +291,17 @@ spec:
...
+      annotations:
+        description: Prometheus {{$labels.namespace}}/{{$labels.pod}} query API has
+          less than 20% available capacity in its query engine for the last 15 minutes.
+        runbook_url: https://runbooks.prometheus-operator.dev/runbooks/prometheus/prometheushighqueryload
+        summary: Prometheus is reaching its maximum capacity serving concurrent requests.
+      expr: |
+        avg_over_time(prometheus_engine_queries{job="prometheus-k8s",namespace="monitoring"}[5m]) / max_over_time(prometheus_engine_queries_concurrent_max{job="prometheus-k8s",namespace="monitoring"}[5m]) > 0.8
+      for: 15m
+      labels:
+        severity: warning
```
Comment (ralgozino) [inline katalog/prometheus-operated/prometheusRule.yaml:304]: nice!

File: katalog/prometheus-operator/README.md (hunk 6 above)
Comment (ralgozino) [reply katalog/prometheus-operator/README.md:42]: mental note: we'll need to keep this in mind for the next releases of KFD, we can't update monitoring for KFD 1.23.x

**Review by ralgozino — CHANGES_REQUESTED** [review]
besides what @nutellinoit said, the rest is looking good.

File: README.md [hunk 9]
```diff
@@ -76,10 +76,11 @@ Please refer the individual package documentation for further details.
...
 | Kubernetes Version |   Compatibility    |                        Notes                        |
 | ------------------ | :----------------: | --------------------------------------------------- |
-| `1.21.x`           | :white_check_mark: | No known issues                                     |
-| `1.22.x`           | :white_check_mark: | No known issues                                     |
-| `1.24.x`           | :white_check_mark: | No known issues                                     |
+| `1.21.x`           |                    | No known issues                                     |
+| `1.22.x`           |                    | No known issues                                     |
+| `1.23.x`           |                    | No known issues                                     |
 | `1.24.x`           | :white_check_mark: | No known issues                                     |
+| `1.25.x`           | :white_check_mark: | No known issues                                     |
```
Comment (nutellinoit) [inline README.md:83]: Revert this change, let's leave the readme as it is for the moment

**Review by nutellinoit — CHANGES_REQUESTED** [review]
One minor change on the main readme 👇 

otherwise LGTM

**Review by nutellinoit — APPROVED** [review]
LGTM!

**Review by ralgozino — APPROVED** [review]
lgtm
